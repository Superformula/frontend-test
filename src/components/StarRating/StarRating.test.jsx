import React from 'react';
import { shallowWithTheme } from '../../helpers/tests/withTheme';

import { StarRating } from '..';

describe('StarRating Component', () => {
  let component;
  it('renders a 3.5 star rating', () => {
    component = shallowWithTheme(<StarRating rating={3.5} />).dive();
    expect(component).toMatchSnapshot();
  });
  it('renders a 5 star rating', () => {
    component = shallowWithTheme(<StarRating rating={5} />).dive();
    expect(component).toMatchSnapshot();
  });
  it('renders a 0 star rating', () => {
    component = shallowWithTheme(<StarRating rating={0} />).dive();
    expect(component).toMatchSnapshot();
  });
});
