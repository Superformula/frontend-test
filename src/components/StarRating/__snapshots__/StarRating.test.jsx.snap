// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating Component renders a 0 star rating 1`] = `
<styled.div
  my={2}
>
  <styled.div
    key="star-1"
  >
    <SvgStarEmpty />
  </styled.div>
  <styled.div
    key="star-2"
  >
    <SvgStarEmpty />
  </styled.div>
  <styled.div
    key="star-3"
  >
    <SvgStarEmpty />
  </styled.div>
  <styled.div
    key="star-4"
  >
    <SvgStarEmpty />
  </styled.div>
  <styled.div
    key="star-5"
  >
    <SvgStarEmpty />
  </styled.div>
</styled.div>
`;

exports[`StarRating Component renders a 3.5 star rating 1`] = `
<styled.div
  my={2}
>
  <styled.div
    key="star-1"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-2"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-3"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-4"
  >
    <SvgStarHalf />
  </styled.div>
  <styled.div
    key="star-5"
  >
    <SvgStarEmpty />
  </styled.div>
</styled.div>
`;

exports[`StarRating Component renders a 5 star rating 1`] = `
<styled.div
  my={2}
>
  <styled.div
    key="star-1"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-2"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-3"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-4"
  >
    <SvgStarFull />
  </styled.div>
  <styled.div
    key="star-5"
  >
    <SvgStarFull />
  </styled.div>
</styled.div>
`;
